### DISCUSSION TITLE
Geometries conversion

### DISCUSSION THREAD

**Discussion by BenjaminDeleze** [cyberbotics/webots#3515] [Ideas]
I finished to convert the geometries (cylinder, capsule, cone, plane and elevation grid) to be z-aligned.

But I have some doubts.

This is what I have done so far:
![before_all](https://user-images.githubusercontent.com/25938827/127283669-f8b0b6e6-93c8-4b27-a537-23fca33a9738.png)

For both images, it is an ENU world. The geometries on the first image are rotated of (1, 0, 0, 1.57) to appear correctly.

I made the modification such that it is the simplest to pass from the old geometries to the new one (only one rotation around one axis)

But I am wondering if it is correct for the mapping of textures. I did not find a convention about where to begin to apply the texture when mapping it to a cylinder,... In the webots documentation it is often `the texture wraps counterclockwise (from above) starting at the back of the cone/cylinder/capsule. The texture has a vertical seam at the back in the yz plane.`. So now it is the same but starting at the front.
But is it correct from an FLU point of view? shouldn't the vertical seam be at back/front in the xz plante ?

My other question concerns elevation grid.
I converted it in the same way as other, such that it is the easiest to switch from old to new:
![before](https://user-images.githubusercontent.com/25938827/127284816-1f7f4e6a-8f99-47fe-9fbf-28679414797c.png)

But as you can see the elevation grid is now in the -yx quarter instead of yx one and I am not sure it is really good. But fixing that would increase the complexity of switching from old to new system

**Comment by stefaniapedrazzi** [comment]
> But as you can see the elevation grid is now in the -yx quarter instead of yx one and I am not sure it is really good. But fixing that would increase the complexity of switching from old to new system

In any case for users it will be complicated to switch to the new system.
We should go for the best solution that makes more sense on the long term and move the elevation grid in the yx quarter.

> But I am wondering if it is correct for the mapping of textures. I did not find a convention about where to begin to apply the texture when mapping it to a cylinder,... In the webots documentation it is often the texture wraps counterclockwise (from above) starting at the back of the cone/cylinder/capsule. The texture has a vertical seam at the back in the yz plane.. So now it is the same but starting at the front.
But is it correct from an FLU point of view? shouldn't the vertical seam be at back/front in the xz plante ?

Previously texture mapping was following the VRML convention but we should no longer stick to it.
Did you for example check what is the default behavior in Blender?

**Reply by BenjaminDeleze** [reply]
> We should go for the best solution that makes more sense on the long term and move the elevation grid in the yx quarter.

I agree

**Comment by BenjaminDeleze** [comment]
I just checked in Blender and obtained that:
![blender](https://user-images.githubusercontent.com/25938827/127297938-7ef7a3de-f251-4ac4-99cb-6afc1abda71f.png)

So it seems correct as it is now.
